src/frontend/sharedComponents/SaveButton.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import {View} from 'react-native';
+import {UIActivityIndicator} from 'react-native-indicators';
+import {IconButton} from './IconButton';
+import SaveCheck from '../images/CheckMark.svg';
+export const SaveButton = ({
+  onPress,
+  isLoading,
+}: {
+  onPress: () => void;
+  isLoading: boolean;
+}) => {
+  return isLoading ? (
+    <View style={{marginRight: 10}}>
```

Comment:
nit: consumer of this component should be responsible for the spacing here

Reply:
the consuming component doesn't have the granularity control of changing the styling of only the loading state. Unless we refactor it, I'm not sure there is a way to have the header change the styling of only the loading state.

This actually is a problem with how the icon button was implemented (which was ported over from mobile). It has a width of 60, and then is centered. In this case it makes it look like it has a margin right of 10. So if we don't add the margin right of 10 to the loader, it causes it be in a different place from the button. So this margin was a hacky way of fixing that.

I think in an ideal world we would refactor the icon button, so that the consumer can deal with the margin/width. But the icon button is used in many places, so I think it is out of scope to do that in this PR.